### PR TITLE
WD-11512 Refresh /download/risc-v page

### PR DESCRIPTION
**Pull request by immortalcodes**
## Done

Webpage refresh for the page /download/risc-v
## QA
- Check out this feature branch
- Run the site using the command dotrun
- View the site locally in your web browser at: http://0.0.0.0:8010/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- [Copydoc](https://docs.google.com/document/d/1SPeefGBJBh99XWEx0newZHPxp56mZuaElq1YtUzfY4M/edit)
- [Demo](https://cn-ubuntu-com-725.demos.haus/download/risc-v)
## Issue / Card
[WD-11512](https://warthogs.atlassian.net/browse/WD-11512)

[WD-11512]: https://warthogs.atlassian.net/browse/WD-11512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by carkod** [comment]
Design looks a bit weird... the bar when you click on it goes to the bottom? Please see this page for reference https://ubuntu.com/download/risc-v

**Comment by carkod** [comment]
Also there should be some kind of separation between the section and the previous content. But I assume you didn't get design for this. This should have some kind of design you can follow, @lyubomir-popov can you help with some pointers?

File: templates/download/risc-v/tab-5.html [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
+                    <p>
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/{{ releases.lts.full_version }}/release/ubuntu-{{ releases.lts.full_version }}-preinstalled-server-riscv64+unmatched.img.xz">下载 {{ releases.lts.full_version }}</a>
+                    </p>
+                </li>
+                <li class="p-list__item">
+                    <h3 class="p-heading--4">Ubuntu Server在线安装程序</h3>
+                    <p>
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/24.04/release/ubuntu-24.04-live-server-riscv64.img.gz">[ 下载 24.04 在线安装程序]</a>
```
Comment (carkod) [inline templates/download/risc-v/tab-5.html:26]: Can you remove the brackets ("[")? These are meant to be indicators of a `p-button` in the copy doc. Thanks

File: templates/download/risc-v/tab-4.html [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
+                    <p>
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/noble/release/ubuntu-{{ releases.lts.full_version }}-preinstalled-server-riscv64+unmatched.img.xz?_gl=1*1j7h704*_gcl_au*OTM2OTYyNzYzLjE3MTI3NjA1NDE.&_ga=2.84786457.1416727561.1716479821-1146325859.1716479821">下载 {{ releases.lts.full_version }}</a>
+                    </p>
+                </li>
+                <li class="p-list__item">
+                    <h3 class="p-heading--4">Ubuntu Server在线安装程序</h3>
+                    <p>
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/noble/release/ubuntu-24.04-live-server-riscv64.img.gz?_gl=1*12l39oz*_gcl_au*OTM2OTYyNzYzLjE3MTI3NjA1NDE.&_ga=2.46111652.1416727561.1716479821-1146325859.1716479821">[ 下载 24.04 在线安装程序]</a>
```
Comment (carkod) [inline templates/download/risc-v/tab-4.html:21]: Remove `[` please. THese are indicators of `p-button`

File: templates/download/risc-v/index.html [hunk 3]
```diff
@@ -1,74 +1,112 @@
...
+    <div class="row">
+      <div class="col-4 col-medium-2">
+        <div class="p-tabs--vertical">
+          <div class="p-tabs__list js-tabbed-content"
+               role="tablist"
+               aria-label="Install RISC-V">
+            <div class="p-tabs__item">
+              <button class="p-tabs__link"
+                      role="tab"
+                      aria-selected="false"
```
Comment (carkod) [inline templates/download/risc-v/index.html:50]: This is the first element selected? In which case I believe it should be `true`

File: templates/download/risc-v/index.html [hunk 4]
```diff
@@ -1,74 +1,112 @@
...
     </div>
-  </div>
-</section>
-
-<script src="/static/js/src/tabbedContent.js"></script>
-
-{% endblock content %}
\ No newline at end of file
+  </section>
+  <script src="/static/js/src/tabbedContent.js"></script>
```
Comment (carkod) [inline templates/download/risc-v/index.html:111]: Optionally (I think it'd be good learning) would you be able change this to use `versioned_static` for this? e.g. https://github.com/canonical/ubuntu.com/blob/f469d7ec8d1e83207e8221bb1dff42477ffb94ed/templates/ceph/install.html#L277

File: templates/download/risc-v/tab-6.html [hunk 5]
```diff
@@ -1,38 +1,32 @@
...
+            <h3 class="p-heading--4">Ubuntu Server预安装镜像</h3>
+            <p>
+                <a class="p-button--positive"
+                   href="https://cdimage.ubuntu.com/releases/{{ releases.lts.full_version }}/release/ubuntu-{{ releases.lts.full_version }}-preinstalled-server-riscv64+licheerv.img.xz">下载 {{ releases.lts.full_version }}</a>
+            </p>
+        </div>
+        <div class="col-4 u-hide--medium u-hide--small u-align--center">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/b151abd7-Sipeed+LicheeRV+-+Edited-2.png",
+            alt="",
```
Comment (carkod) [inline templates/download/risc-v/tab-6.html:18]: Usually UX would ask us to fill this up with the name of the board.

File: templates/download/risc-v/tab-7.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/24.04/release/ubuntu-24.04-live-server-riscv64.img.gz">[ 下载 24.04 在线安装程序]</a>
+                    </p>
+                </li>
+            </ul>
+        </div>
+        <div class="col-4 u-hide--medium u-hide--small u-align--center">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/470a2398-StarFive+VisionFive+Board-NoBg.png",
+            alt="",
```
Comment (carkod) [inline templates/download/risc-v/tab-7.html:29]: Usually UX would ask us to fill this up with the name of the board.

File: templates/download/risc-v/tab-5.html [hunk 7]
```diff
@@ -1,32 +1,48 @@
...
+                    </p>
+                </li>
+            </ul>
+        </div>
+        <div class="col-4 u-hide--medium u-hide--small u-align--center">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/37c094a9-SiFive+Unmatched+Board.jpg",
+            alt="",
```
Comment (carkod) [inline templates/download/risc-v/tab-5.html:34]: Usually UX would ask us to fill this up with the name of the board.

File: templates/download/risc-v/tab-4.html [hunk 8]
```diff
@@ -1,32 +1,43 @@
...
+                    </p>
+                </li>
+            </ul>
+        </div>
+        <div class="col-4 u-hide--medium u-hide--small u-align--center">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/308d48cd-Qemu-logo.png",
+            alt="",
```
Comment (carkod) [inline templates/download/risc-v/tab-4.html:29]: Usually UX would ask us to fill this up with the name of the board.

**Comment by immortalcodes** [comment]
> Design looks a bit weird... the bar when you click on it goes to the bottom? Please see this page for reference https://ubuntu.com/download/risc-v

For some reason it's not going sideways as it should also when you again click it somewhere the problem seems to go away

File: static/js/src/tabbedContent.js [hunk 9]
```diff
@@ -83,3 +83,29 @@
...
+  const boards = document.querySelectorAll(`[role=tabpanel]`);
+  const dropdownSelect = document.getElementById('boardSelect');
+
+  if (dropdownSelect) {
+    dropdownSelect.addEventListener('change', () => {
+      selectBoard();
+    });
+  } else {
+    console.error('dropdownSelect element not found');
+  }
```
Comment (carkod) [inline static/js/src/tabbedContent.js:99]: Is this logging necessary? Can you investigate in what circumstances `dropdownSelect` is not defined and cover as much as possible, say for example you may want to add an [`onload`](https://stackoverflow.com/questions/1033398/how-to-execute-a-function-when-page-has-fully-loaded) listener.

This logging sounds like it should be there mainly for debugging purposes.

Comment (immortalcodes) [reply static/js/src/tabbedContent.js:99]: Leaving it blank was causing linting issues so I logged in the error, but yeah probably added onload could be a way.

**Comment by carkod** [comment]
The only issue left I see visually is the underline of the tabs. You may need to remove or add some styling to make that work. Let us know if you need help. Else, looks good to me, If Lyubo is fine with it.

File: templates/download/risc-v/tab-7.html [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            <ul class="p-list--divided">
+                <li class="p-list__item">
+                    <h3 class="p-heading--4">Ubuntu Server预安装镜像</h3>
+                    <p>
+                        <a class="p-button--positive"
+                           href="https://cdimage.ubuntu.com/releases/{{ releases.lts.full_version }}/release/ubuntu-{{ releases.lts.full_version }}-preinstalled-server-riscv64+visionfive.img.xz">下载 {{ releases.lts.full_version }}</a>
+                    </p>
+                </li>
+                <li class="p-list__item">
+                    <h3 class="p-heading--4">Ubuntu Server在线安装程序</h3>
```
Comment (carkod) [inline templates/download/risc-v/tab-7.html:18]: Indentation looks a bit off on this file. You may want to talk to Akbar maybe to get that sorted to do it automatically?

**Comment by carkod** [comment]
Looks like you have inconsistent indentation in all the `tab-*` files, can you also fix that?

**Comment by immortalcodes** [comment]
> Looks like you have inconsistent indentation in all the `tab-*` files, can you also fix that?

The indedation  is not me, it's DJlint. I thought it makes hierarchy of tags more clear

**Comment by carkod** [comment]
> > Looks like you have inconsistent indentation in all the `tab-*` files, can you also fix that?
> 
> The indentation is not me, it's DJ-lint. I thought it makes hierarchy of tags more clear

ok, could be because there is no djlint file. Can you add https://github.com/canonical/ubuntu.com/blob/main/.djlintrc#L7 to this project? Should help?

As you can see indentation should be 2 spaces.

**Comment by immortalcodes** [comment]
> > > Looks like you have inconsistent indentation in all the `tab-*` files, can you also fix that?
> > 
> > 
> > The indentation is not me, it's DJ-lint. I thought it makes hierarchy of tags more clear
> 
> ok, could be because there is no djlint file. Can you add https://github.com/canonical/ubuntu.com/blob/main/.djlintrc#L7 to this project? Should help?
> 
> As you can see indentation should be 2 spaces.

Yeah sure, here is the PR, quick approve please
https://github.com/canonical/cn.ubuntu.com/pull/726

File: templates/download/risc-v/index.html [hunk 11]
```diff
@@ -2,73 +2,129 @@
...
+  <section class="p-strip--suru-topped">
+    <div class="row u-vertically-center">
+      <div class="col-6">
+        <h1>下载适用于RISC-V平台的Ubuntu</h1>
+        <h2 class="p-heading--3">Ubuntu——现已支持多个RISC-V平台</h2>
+        <p>在您喜爱的RISC-V开发板上运行Ubuntu。</p>
+        <p>
+          选择与您的硬件相匹配的操作系统镜像，将系统镜像刷入到 SD/microSD 卡上，再将其加载到您的板上即可开始体验Ubuntu。
+          <br />
+          所有镜像都是基于64位Ubuntu服务器版而为开发者预览所打造。
```
Comment (carkod) [inline templates/download/risc-v/index.html:24]: Looking at the copy doc, this should be a separate paragraph, not a break.

```suggestion
          <p>所有镜像都是基于64位Ubuntu服务器版而为开发者预览所打造。</p>
```

**Review by carkod — APPROVED** [review]
Please, address the 2 comments. Other than that LGTM! Thank you.

**Pull request merged**